### ISSUE TITLE
compatibility break: unwanted stderr output 

### ISSUE THREAD

**Issue by dk** [garu/Data-Printer#143]
Hello,

I was upgrading old modules and there's some breakage of the following code between 0.35 and 0.40:

```
use Data::Printer alias => 'pp', use_prototypes => 0;
my %foo = (1,2);
$_ = pp(\%foo);
```

while 0.35 detects correctly that we need formatted data as a result and there's nothing printed, as expected, 0.40 dumps the data also to the stderr and returns just value of 1, which messes up everything

**Comment by garu** [comment]
Hi @dk! Thank you for using Data::Printer and for taking the time to submit this issue. Also, apologies for taking so long to respond. You're experiencing the side-effect of a backwards incompatible change made on 0.36 that altered the default `return_value` option from "dump" to "pass".

TL;DR: to keep the old behaviour just set the property back, either by:

```perl
    use Data::Printer alias => 'pp', use_prototypes => 0, return_value => 'dump';
    my %foo = (1,2);
    $_ = pp(\%foo);
```
or by setting that in your .dataprinter file.

The default return type is now "pass", which will print to STDERR while also returning the data structure (in your case it's returning \%foo).

Because this is a debugging tool please note we do not make backwards-incompatible changes lightly: there was a pretty long discussion here (see #16) and it was documented [here](https://metacpan.org/pod/Data::Printer#Return-Value).

Hope this helps! Cheers!
